### DISCUSSION TITLE
Update or Installation from laberning's v1beta to main

### DISCUSSION THREAD

**Discussion by jhlee98y** [JaapvanEkris/openrowingmonitor#17] [Q&A] [answered]
Hi, 
How can I update or install v1beta_updates version from  laberning's v1beta ? 

A month ago, I installed laberning's v1beta on RPI 3B+ and checked  some services working properly. 
( My rowing machine is CHINA AIR ROWER as C2 clone. And also I installed 2 magnets on the flywheel and hall effect sensor.) 

Actually I'm newbie for GIT. So I want to know how to update to your v1beta_updates version. 

Thanks in advance.

**Comment by JaapvanEkris** [comment]
What I did was open the install script and replace every occurance of laberning to JaapvanEkris, and change the branch name to v1beta_updates . Then you can rerun the install script (make sure to save your config.js). Then it should install.

We hope to update the V1Beta branch soon with V1Beta_Updates, but some bugs still need to be squashed and perhaps some functinality will be added before that happends.

**Reply by jhlee98y** [reply]
Thank you for your reply. 

Today, I tried some attempts for update or clean installation.
First, I simply replaced Github remote repository "laberning to JaapvanEkris" in  /opt/openrowingmonitor/bin/updateopenrowingmonitor.sh file.  And I executed this file and result are 

pi@rowingmonitor:~ $ sudo /opt/openrowingmonitor/bin/updateopenrowingmonitor.sh -b v1beta_updates
Update script for Open Rowing Monitor

Checking for the existence of branch "v1beta_updates"...
Do you want to switch from branch "v1beta" to branch "v1beta_updates"? [Y/n] y
Switching to branch "v1beta_updates"...
Stopping Open Rowing Monitor...
Switching Open Rowing Monitor to branch "v1beta_updates"...
fatal: 'origin/v1beta_updates' is not a commit and a branch 'v1beta_updates' cannot be created from it 

I tried some fetch and clone your v1beta_updates repository. But I failed !!

So, for the clean install, I reinstalled the Rip OS 64bit Lite version and setup the config , update OS and install git. 
 And also, I clone  your v1beta_updates repository in pi account  and replaced remote git url  "laberning to JaapvanEkris"  in install.sh file.  
Finally I executed install.sh and  however, stopped  installation for  openrowingmonitor step. ( upto Node.js step,  successfully installed.)

I would like you to check my mistake and attempts.  
And give some detailed installation guide because NOT good at Linux system especially git operation.
Thanks in advance.

**Reply by Gordon-Shumway2** [reply]
Re-install the OS, download 'https://github.com/JaapvanEkris/openrowingmonitor/blob/v1beta_updates/install/install.sh', replace  every occurance of "laberning" with "JaapvanEkris" and change the branch name(s) from 'main' to 'v1beta_updates' within lines 158 - 168. Then run 'install.sh' - you might have to run 'chmod 755 install.sh' to make it executable. 

These steps worked for me. If they do not work for you, you might want to upload the last 30 lines of the command's output.

Update January 2024: similar changes should be carried out for the update scripts within /opt/openrowingmonitor/bin/.

**Reply by jhlee98y** [reply]
Thank you for your kind reply & guide. 
Today, I followed your guide especially replaced repository name 'main' to 'v1beta_updates'. 
I completed update code  and config, Finally, I got the rowing monitor screen on my old ipad2 screen( with NOMACHINE).   
The H10 heart rate with battery status is so GREAT!! 
 
I really appreciate your update version work and thank you again.

![OpenRowingMonitor_screen](https://user-images.githubusercontent.com/7874591/235098770-1df6a229-9ae7-4b3c-9dee-20dae823b7a7.jpg)

**Reply by JaapvanEkris** [reply]
I tried upgrading to a newer Node.js version, which seems to run into some dependency issues.

I'm working on that one. I'm also considering to change the published install script in such a way that the dependency of labernings repo is removed and we incorporate all above changes into the normal install script.

**Comment by JaapvanEkris** [comment]
Dear all: I improved the installation procedure by modifying the install script for you. Just use:
```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/JaapvanEkris/openrowingmonitor/main/install/install.sh)"
```

**Reply by Techtobi83** [reply]
This is a huge improvement for beginners like me, thank you. 

Confirm me: Simply install the OS on a SD card, then execute your command (or have it executed) and you're completly done and have v1beta_updates branch running

**Comment by JaapvanEkris** [comment]
> Confirm me: Simply install the OS on a SD card, then execute your command (or have it executed) and you're completly done and have v1beta_updates branch running

That is what the test show us. Please note: we still have a problem in using Bookwork (there is a conflict in Node-versions in the Pi Zero branch), so you must use Buster or Bullseye.

**Comment by JaapvanEkris** [comment, marked as answer]
We fixed the issues with Bookworm, and it now can be installed on newer Raspberry Pi OS installs as well. I renamed the current branch to main (was V1Beta_Updates, but as we expect to move to V1 and beyond it was a bit too fixated...).

To install the latest stable version of OpenRowingMonitor on the Raspberry Pi Zero2/3/4/5, you can follow these instructions: https://github.com/JaapvanEkris/openrowingmonitor/blob/main/docs/installation.md

To install the latest stable version of OpenRowingMonitor on a Raspberry Pi Zero, please follow these instructions: https://github.com/JaapvanEkris/openrowingmonitor/tree/v1beta_updates_Pi_Zero_W Please be aware: due to circumstances beyond our control, support for this platform will be dropped in April 2025 (see https://github.com/JaapvanEkris/openrowingmonitor/discussions/33).
